packages/cyberstorm/src/components/TextInput.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { MouseEventHandler, PropsWithChildren, ReactNode } from "react";
+import styles from "./componentStyles/TextInput.module.css";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCircleX } from "@fortawesome/pro-regular-svg-icons";
+
+export interface TextInputProps {
+  disabled?: boolean;
+  leftIcon?: ReactNode;
+  onClear?: MouseEventHandler<HTMLButtonElement>;
+  onChange(event: React.ChangeEvent<HTMLInputElement>): void;
+  onLeftIconClick?: MouseEventHandler<HTMLButtonElement>;
+  onRightIconClick?: MouseEventHandler<HTMLButtonElement>;
+  placeholder?: string;
+  rightIcon?: ReactNode;
+  size?: "default" | "small";
+  value: string;
+}
+
+const getStyle = (scheme: TextInputProps["size"] = "default") => {
```

Comment:
I think `getStyle` is used in other components to get the colorScheme styles. Maybe this should be `getSize`?